kotsadm/pkg/updatechecker/updatechecker.go

```diff
@@ -0,0 +1,210 @@
+package updatechecker
+
+import (
+	"database/sql"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/kots/kotsadm/pkg/app"
+	"github.com/replicatedhq/kots/kotsadm/pkg/kotsutil"
+	"github.com/replicatedhq/kots/kotsadm/pkg/license"
+	"github.com/replicatedhq/kots/kotsadm/pkg/logger"
+	"github.com/replicatedhq/kots/kotsadm/pkg/task"
+	"github.com/replicatedhq/kots/kotsadm/pkg/upstream"
+	"github.com/replicatedhq/kots/kotsadm/pkg/version"
+	kotspull "github.com/replicatedhq/kots/pkg/pull"
+	cron "github.com/robfig/cron/v3"
+	"go.uber.org/zap"
+)
+
+// jobs maps app ids to their cron jobs
+var jobs map[string]*cron.Cron
+var mtx sync.Mutex
+
+// Start will start the update checker
+// the frequency of those update checks are app specific and can be modified by the user
+func Start() error {
+	logger.Debug("starting update checker")
+
+	appsList, err := app.ListInstalled()
+	if err != nil {
+		return errors.Wrap(err, "failed to list installed apps")
+	}
+
+	for _, a := range appsList {
+		if a.IsAirgap {
+			continue
+		}
+		if err := Configure(a.ID); err != nil {
+			return errors.Wrapf(err, "failed to configure app %s", a.Slug)
+		}
+	}
+
+	return nil
+}
+
+// Configure will check if the app has scheduled update checks enabled and:
+// if enabled, and cron job was NOT found: add a new cron job to check app updates
+// if enabled, and a cron job was found, update the existing cron job with the latest cron spec
+// if disabled: stop the current running cron job (if exists)
+// returns a "not supported" error for airgap applications
+func Configure(appID string) error {
+	a, err := app.Get(appID)
+	if err != nil {
+		return errors.Wrap(err, "failed to get app")
+	}
+
+	if a.IsAirgap {
+		return errors.New("airgap scheduled update checks are not supported")
```

Comment:
this will cause an error to be logged.  this is not an error.

Reply:
will or should?